jupyterhub/app.py

```diff
@@ -2265,6 +2267,9 @@ def _log_cls(name, cls):
         self.init_tornado_settings()
         self.init_handlers()
         self.init_tornado_application()
+        HUB_STARTUP_DURATION_SECONDS.observe(
+            time.perf_counter() - hub_startup_start_time
+        )
 
         # init_spawners can take a while
```

Comment:
Since ``init_spawners`` can take a while and because we're already logging its duration I believe it would be nice to also expose this duration as a prometheus metric (*INIT_SPAWNERS_DURATION_SECONDS*). This way, if hub startup takes long, we can check INIT_SPAWNERS_DURATION_SECONDS and see if this is the reason. What do you think?

Reply:
Because spawners initialization is part of hub initialization, I think separating them might cause a bit of confusion :thinking:.
Also, what do you think of having *HUB_STARTUP_DURATION_SECONDS* reflect not only the initialization time, but also the entire time it takes to have the Hub up and running, i.e. the sum of init time and start time?

Reply:
My apologies for the late response.
I've exposed `INIT_SPAWNERS_DURATION_SECONDS` metric as you suggested.

Regarding `HUB_STARTUP_DURATION_SECONDS` reflecting time taken for Hub to be up and running - I don't think I have enough context/knowledge to make that call. Can you please share what's normally considered as `Hub Startup` - the initialization, or Hub running?

If it seems confusing, perhaps we can rename/split the metric:
* HUB_INITIALIZATION_DURATION_SECONDS - Initialization time
* HUB_STARTUP_DURATION_SECONDS - Initialization & Hub up and running time

Please suggest which approach to take.

Reply:
Since this has proven not to be very straightforward, maybe @betatim, can help us decide. @betatim, what do you think, **hub startup time** should be hub initialization time or hub init + hub start time?